### PR TITLE
improve type hints

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Add type hints

### Linked issues
None

### Functionality
None

### Tests
- [x] ran unit tests

### CODE REVIEW

File: src/databricks/labs/ucx/assessment/azure.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _get_relevant_service_principals(self) -> list[AzureServicePrincipalInfo]:
 
         # list all relevant service principals in jobs
         all_jobs = list(self._ws.jobs.list(expand_tasks=True))
-        all_clusters_by_id = {c.cluster_id: c for c in self._ws.clusters.list()}
+        all_clusters_by_id = {c.cluster_id or "": c for c in self._ws.clusters.list()}
```
Comment (nfx) [inline src/databricks/labs/ucx/assessment/azure.py:76]: skip the cluster

File: src/databricks/labs/ucx/assessment/jobs.py [hunk 2]
```diff
@@ -79,7 +83,7 @@ def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
 
     def _crawl(self) -> Iterable[JobInfo]:
         all_jobs = list(self._ws.jobs.list(expand_tasks=True))
-        all_clusters = {c.cluster_id: c for c in self._ws.clusters.list()}
+        all_clusters = {c.cluster_id or "": c for c in self._ws.clusters.list()}
```
Comment (nfx) [inline src/databricks/labs/ucx/assessment/jobs.py:86]: skip the cluster

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/assessment/azure.py (hunk 1 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/assessment/azure.py]: done

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**